src/main/webapp/utilities/MapsInterface.js

```diff
@@ -57,10 +57,18 @@ class MapInterface {
     async converToCoordinates(text, fetch = windows.fetch) {
         const url = `https://maps.googleapis.com/maps/api/geocode/json?key=${this.API_KEY}&address=${text}`; // Maps JS API can also be used
 
-        const response = await fetch(url);
-        const json = await response.json();
+        try {
+            const response = await fetch(url);
+            const json = await response.json();
 
-        return (json.status === 'OK' ? json.results[0].geometry.location : {}); // TODO
+            if(json.status === 'OK') {
+                return json.results[0].geometry.location;
+            } else {
+                throw 'Error with the API';
```

Comment:
Please make it more verbose, add json.status. if there is something more than status please log it to console. I believe we also discussed that it might return not found and it should be handled separately.

Reply:
I thought in case it returns not found, then we can still return {} (I don't remember sth else). But for sure I can also return the status!

Reply:
Updated it! I think now it's better!